src/Storages/System/StorageSystemTables.cpp

```diff
@@ -102,11 +123,20 @@ ColumnPtr getFilteredTables(const ASTPtr & query, const ColumnPtr & filtered_dat
             continue;
 
         for (auto table_it = database->getTablesIterator(context); table_it->isValid(); table_it->next())
```

Comment:
I checked `getTablesIterator` implementation of PostgreSQL and MySQL one more time. In both databases we connect to PostgreSQL/MySQL to get a list of tables, but in PostgreSQL we do it in try/catch and don't throw an exception (there is even a comment that it's done because of system.tables table), but in MySQL I don't see this logic, so we can throw an exception there.